usaspending_api/etl/es_etl_helpers.py

```diff
@@ -249,6 +252,45 @@ def streaming_post_to_es(client, chunk, index_name, job_id=None):
     return success, failed
 
 
+def put_alias(client, index, alias_name, award_type_codes):
+    alias_body = {'filter': {'terms': {"type": award_type_codes}}}
+    client.indices.put_alias(index, alias_name, body=alias_body)
+
+
+def swap_aliases(client, index):
+    client.indices.refresh(index)
+    # add null values to contracts alias
+    if client.indices.get_alias(index, '*'):
+        printf({'msg': 'Removing old aliases for index "{}"'.format(index),
+                'job': None, 'f': 'ES Alias Drop'})
+        client.indices.delete_alias(index, '_all')
+
+    indices_to_award_types['contracts'] += ('NULL',)
+    alias_patterns = settings.TRANSACTIONS_INDEX_ROOT + '*'
+
+    try:
+        old_indices = client.indices.get_alias('*', alias_patterns).keys()
+        for old_index in old_indices:
+            client.indices.delete_alias(old_index, '_all')
+            client.indices.close(old_index)
+            printf({'msg': 'Removing aliases & closing "{}"'.format(old_index),
+                    'job': None, 'f': 'ES Alias Drop'})
+    except Exception as e:
+        printf({'msg': 'ERROR: no aliases found for {}'.format(alias_patterns),
+                'f': 'ES Alias Drop'})
+
+    for award_type, award_type_codes in indices_to_award_types.items():
+        alias_name = '{}-{}'.format(settings.TRANSACTIONS_INDEX_ROOT, award_type)
+        printf({'msg': 'Putting alias "{}" with award codes {}'.format(alias_name, award_type_codes),
+                'job': '', 'f': 'ES Alias Put'})
+        put_alias(client, index, alias_name, award_type_codes)
+
+    index_settings = dict(refresh_interval='1s', number_of_replicas=1)
```

Comment:
Can these settings be moved to another location to make it easy to see what the "final settings" will be for the index? One idea is to have an "index-settings" which are applied to the index before indexing (initial and reload) and then revert to the final settings once the data has been loaded in and the script is moving on